### PR TITLE
Add `Unknown` type to `TaskInfoType`

### LINKED ISSUE #385: Add generic type to `TaskInfoType` to handle new `Task` type
**Description**
All types of tasks are grouped in an enum `TaskInfoType`
```c#
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum TaskInfoType
    {
        IndexCreation,
        IndexUpdate,
        ...
        IndexSwap
    }
```
But if a new [task type is created](https://github.com/meilisearch/meilisearch/issues/3162) there will be a break.

**Suggestion**
To avoid breaking when a [new `Task` type is created](https://github.com/meilisearch/meilisearch/issues/3162), it would be great to create a generic type such as `unknown` that will handle all types that do not belong to the existing enum.

### PR DESCRIPTION
**Pull request by postmeback**
# Pull Request

## Related issue
Fixes #385 

## What does this PR do?
- Currently, it adds the "Unknown" type to the enum. It requires more tasks to do

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by curquiza — CHANGES_REQUESTED** [review]
As discussed, can you add tests? 😊

**Comment by postmeback** [comment]
Yeah, I will. This PR will get updated soon.

**Comment by postmeback** [comment]
> Yeah, I will. This PR will get updated soon.

I have updated this PR, to the best of my capabilities. Please review and provide feedback.

**Comment by postmeback** [comment]
Seems like the build failed, Can you guide me on where to look?
@curquiza

**Comment by curquiza** [comment]
@postmeback 
The tests [on `main` are green](https://github.com/meilisearch/meilisearch-dotnet/actions/runs/9776036741), so it's something you added that broke the CI.
Please, run the tests locally, you will have a better view of what is happening to fix them.

**Comment by postmeback** [comment]
Well, I tried to run the tests, and below is the response.I have commented my tests.

image

It seems I am restricted. Any idea on how to get past it ?

**Comment by curquiza** [comment]
@postmeback there is not need for access to run the tests locally.
Did you follow the steps in [the CONTRIBUTING.md](https://github.com/meilisearch/meilisearch-dotnet/blob/main/CONTRIBUTING.md)?

**Comment by postmeback** [comment]
I followed the steps and came up with below conclusion

`CreateUnknownTaskAsync` & `GetTaskWithUnknownTypeAsync` - this two methods have an URI, which I created and not added with API, that's the reason it is failing.

https://github.com/postmeback/meilisearch-dotnet/blob/c2934b51bae7b0f243009af109f26499cb7585a0/src/Meilisearch/MeilisearchClient.cs#L430

https://github.com/postmeback/meilisearch-dotnet/blob/c2934b51bae7b0f243009af109f26499cb7585a0/src/Meilisearch/MeilisearchClient.cs#L438C37-L438C64

Requesting you to have a look.

**Comment by curquiza** [comment]
@postmeback `CreateUnknownTaskAsync` call a route that does not exist in Meilisearch. It cannot work.

Maybe @ahmednfwela can guide you on this, or @brunoocasali who opened the issue.

File: src/Meilisearch/MeilisearchClient.cs [hunk 1]
```diff
@@ -425,5 +425,23 @@ private TaskEndpoint TaskEndpoint()
...
+        }
+
+        public async Task<TaskInfo> GetTaskWithUnknownTypeAsync(GetTaskQuery query, CancellationToken cancellationToken = default)
+        {
+            var response = await _http.PostAsJsonAsync("get-task-with-unknown-type", query, Constants.JsonSerializerOptionsRemoveNulls, cancellationToken: cancellationToken)
+                    .ConfigureAwait(false);
+
+            return await response.Content.ReadFromJsonAsync<TaskInfo>(cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
```
Comment (ahmednfwela) [inline src/Meilisearch/MeilisearchClient.cs]: all of these are unnecessary

File: src/Meilisearch/QueryParameters/GetTaskQuery.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System;
```
Comment (ahmednfwela) [inline src/Meilisearch/QueryParameters/GetTaskQuery.cs]: this new file is unnecessary

File: src/Meilisearch/QueryParameters/UnknownTaskQuery.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using System;
```
Comment (ahmednfwela) [inline src/Meilisearch/QueryParameters/UnknownTaskQuery.cs]: this new file is unnecessary

**Review by ahmednfwela — CHANGES_REQUESTED** [review]
to address the issue, we just need to configure the json serializer used by meilisearch to fallback to the unknown enum value

**Comment by postmeback** [comment]
I have removed the unnecessary files and methods from this PR. 

Also, if you can point to the `json serializer` file/function, which needs to be updated, that would greatly assist. @ahmednfwela

**Comment by ahmednfwela** [comment]
@postmeback we use the default one https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonstringenumconverter?view=net-8.0

if this converter doesn't have a fallback option, we can create a custom one that inherits it

**Comment by ahmednfwela** [comment]
check this [SO question](https://stackoverflow.com/questions/76709269/system-text-json-jsonstringenumconverter-with-custom-fallback-in-case-of-deseria) for similar solutions

**Comment by postmeback** [comment]
Sure, thanks for the suggestion. I will update you soon.

**Comment by postmeback** [comment]
image


With the latest changes and all tests passing

**Comment by postmeback** [comment]
image


I ran the formatter as well.

**Review by ahmednfwela — CHANGES_REQUESTED** [review]
can you please replace the json converter annotation on top of the enum with the new converter

**Comment by postmeback** [comment]
Done.

**Review by ahmednfwela — APPROVED** [review]

**Comment by ahmednfwela** [comment]
bors merge

**Review by curquiza — APPROVED** [review]
missing my approval

bors merge

**Pull request merged**